src/services/telegram/commands/roll.rs

```diff
@@ -0,0 +1,95 @@
+use anyhow::{Context};
+use rand::{Rng, SeedableRng};
+use teloxide::Bot;
+use teloxide::payloads::SendMessageSetters;
+use teloxide::prelude::{Message, Requester};
+use crate::services::telegram::commands::{CommandResult};
+use regex::Regex;
+
+
+
+pub async fn handler(bot: &Bot, message: &Message, roll: &str) -> CommandResult {
+    let mut rng = rand::rngs::SmallRng::from_entropy();
+    if rng.gen_range(1..1001) == 1 {
+        let _reply = bot
+		.send_message(message.chat.id, "🎶 Roll? Rick roll! https://www.youtube.com/watch?v=dQw4w9WgXcQ")
+		.reply_to_message_id(message.id)
+		.await
+		.context("Non è stato possibile inviare la risposta.")?;
+
+	    return Ok(())
+    }
+
+    let re = Regex::new(r#"(?P<qty>[0-9]*)?d(?P<die>[0-9]+)(?P<modifier>[+-]?[0-9]*)?"#);
```

Comment:
La `Regex` è sempre valida, visto che l'hai scritta tu!

Possiamo permetterci di fare:

```suggestion
    let re = Regex::new(r#"(?P<qty>[0-9]*)?d(?P<die>[0-9]+)(?P<modifier>[+-]?[0-9]*)?"#).unwrap();
```

Reply:
Volendo si potrebbe anche rendere `const`, ma lasciamo stare per ora...